query-parser/src/util.rs

```diff
@@ -31,15 +29,15 @@ pub fn value_to_variable(v: &edn::Value) -> Option<Variable> {
 
 /// If the provided slice of EDN values are all variables as
 /// defined by `value_to_variable`, return a Vec of Variables.
-/// Otherwise, return an error.
-pub fn values_to_variables(vals: &[edn::Value]) -> Result<Vec<Variable>, FindParseError> {
+/// Otherwise, return the unrecognized Value.
```

Comment:
I made this change when I split the find parser and the query parser: the `values_to_variables` function itself doesn't know whether this should return a `FindParseError` or a `QueryParseError`.

There's also precedent in the Rust docs themselves:

> Functions return `Result` whenever errors are expected and recoverable. In the `std` crate, `Result` is most prominently used for I/O.
>
> A simple function returning `Result` might be defined and used like so:
>
> ```#[derive(Debug)]
enum Version { Version1, Version2 }

fn parse_version(header: &[u8]) -> Result<Version, &'static str> {
    match header.get(0) {
        None => Err("invalid header length"),
        Some(&1) => Ok(Version::Version1),
        Some(&2) => Ok(Version::Version2),
        Some(_) => Err("invalid version"),
    }
}```

Right below it they give an example of a `Result<i32, i32>`.

I chose `Result` instead of `Option` because this is an operation that can fail (you might choose to use `try`), and it returns the reason why it failed (the value that wasn't a variable).